db.go

```diff
@@ -176,37 +179,53 @@ func (db *sqlDB) PrepareContext(ctx context.Context, query string) (_stmt Stmt,
 
 // Query executes a query that returns rows, typically a SELECT.
 // The args are for any placeholder parameters in the query.
-// Query uses a radonly db as the physical db.
 func (db *sqlDB) Query(query string, args ...interface{}) (*sql.Rows, error) {
-	return db.QueryContext(context.Background(), query, args...)
+	return db.QueryContext(backgroundCtx, query, args...)
 }
 
 // QueryContext executes a query that returns rows, typically a SELECT.
 // The args are for any placeholder parameters in the query.
-// QueryContext uses a radonly db as the physical db.
-func (db *sqlDB) QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error) {
-	rows, err := db.ReadOnly().QueryContext(ctx, query, args...)
-	if isDBConnectionError(err) {
+func (db *sqlDB) QueryContext(ctx context.Context, query string, args ...interface{}) (rows *sql.Rows, err error) {
+	var curDB *sql.DB
+	_query := strings.ToUpper(query)
+	writeFlag := strings.Contains(_query, "RETURNING")
```

Comment:
This may be triggered if there is a column or table named "RETURNING" exists on the query. 

It won't be a big problem because the query would go to the readwrite connection so it would work anyway, but maybe we should see if there is also "INSERT", "UPDATE", or "DELETE" in the query in addition to "RETURNING", to be a little more safe?

Reply:
but same issue might pop up if we try if any of the tokens or conditions match the keys "insert","update", or "delete". 

Eg:

select id, deleted_at from _table__